system_test/aest_hard_fork_SUITE.erl

```diff
@@ -445,6 +506,134 @@ new_node_can_mine_on_old_chain_using_new_protocol(Cfg) ->
+new_node_can_mine_spend_tx_on_old_chain_using_old_protocol(Cfg) ->
+    {_, {_Saver, SavedCfg}} = proplists:lookup(saved_config, Cfg),
+    {_, new_node1} = proplists:lookup(new_node_left_running_with_old_chain, SavedCfg),
+    {_, TopHeight} = proplists:lookup(db_backup_top_height, Cfg),
+    {_, TopHash} = proplists:lookup(db_backup_top_hash, Cfg),
+    %% Sync the chain on new_node3 from new_node1.
+    start_node(new_node3, Cfg),
+    wait_for_height_syncing(TopHeight, [new_node3], {{45000, ms}, {200, blocks}}, Cfg),
+    #{hash := TopHash} = get_block_by_height(new_node3, TopHeight, Cfg),
+    %% Get public key of new_node1 and make sure its balance is 0.
+    PubKey1 = get_public_key(new_node1, Cfg),
+    Balance1 = get_balance(new_node1, PubKey1, Cfg),
+    ?assertEqual(0, Balance1),
+    ct:log("Balance of node ~p with public key ~p is ~p", [new_node1, PubKey1, Balance1]),
+    %% Kill new_node1, it was used just to sync the old chain.
+    aest_nodes:kill_node(new_node1, Cfg),
+    %% Get public key of new_node3 and make sure its balance is 0.
+    PubKey3 = get_public_key(new_node3, Cfg),
+    Balance3 = get_balance(new_node3, PubKey3, Cfg),
+    ?assertEqual(0, Balance3),
+    ct:log("Balance of node ~p with public key ~p is ~p", [new_node3, PubKey3, Balance3]),
+    %% Start mining on new_node3.
+    ok = mock_pow_on_node(new_node3, Cfg),
+    run_erl_cmd_on_node(new_node3, "aec_conductor:start_mining().", "ok", Cfg),
+    %% Check that new_node3 got reward for mining.
+    MinedReward = 100,
+    aest_nodes:wait_for_value({balance, PubKey3, MinedReward}, [new_node3], 10000, Cfg),
+    MinedBalance = get_balance(new_node3, PubKey3, Cfg),
+    ?assert(MinedBalance >= MinedReward),
+    ct:log("Mined balance on ~p with public key ~p is ~p", [new_node3, PubKey3, MinedBalance]),
+    %% Send spend transaction from new_node3 to new_node1.
+    Fee = 5,
+    BalanceToSpend = MinedBalance - Fee,
+    ok = post_spend_tx(new_node3, PubKey1, BalanceToSpend, Fee, Cfg),
+    ct:log("Sent spend tx with balance of ~p from node ~p with public key ~p to node ~p with public key ~p",
+           [BalanceToSpend, new_node3, PubKey3, new_node1, PubKey1]),
+    %% Make sure new_node1 received the balance.
+    aest_nodes:wait_for_value({balance, PubKey1, BalanceToSpend}, [new_node3], 20000, Cfg),
+    ReceivedBalance = get_balance(new_node3, PubKey1, Cfg),
+    ct:log("Balance of node ~p with public key ~p is ~p", [new_node1, PubKey1, ReceivedBalance]),
+    ?assertEqual(BalanceToSpend, ReceivedBalance),
+    %% Sync the chain with the block that includes spend tx on new_node4.
+    start_node(new_node4, Cfg),
+    ok = mock_pow_on_node(new_node4, Cfg),
+    wait_for_height_syncing(TopHeight, [new_node4], {{45000, ms}, {200, blocks}}, Cfg),
+    #{hash := TopHash} = get_block_by_height(new_node4, TopHeight, Cfg),
+    aest_nodes:wait_for_value({balance, PubKey1, ReceivedBalance}, [new_node4], 20000, Cfg),
```

Comment:
This check opens the possibility for the same tx being mined independently by new_node4 (from the test point of view). I suggest the test shall:
* Identify block height and hash on new_node3 known to contain (in chain up to that point - not necessarily in exact block) the spend_tx;
* Rather than this wait for balance, wait for such identified height on new_node4;
* Finally check that at the identified height node new_node4 has the identified hash.